lab3/client/src/views/generator.js

```diff
@@ -0,0 +1,60 @@
+import constants from '../constants.js';
+
+export default function (store) {
+	return class GeneratorComponent extends window.HTMLElement {
+		constructor () {
+			super();
+			this.store = store;
+
+			// TODO: render generator initial view
+			this.innerHTML = this.render();
+
+			// TODO: subscribe to store on change event
+			this.onStateChange = this.handleStateChange.bind(this);
+
+			// TODO: add click event
+			this.addEventListener('click', () => {
+				this.store.dispatch({
+					type: constants.actions.BUY_GENERATOR,
+					payload: {
+						name: this.store.state.generators[this.dataset.id].name,
+						quantity: this.store.state.generators[this.dataset.id].quantity + 1
+					}
+				})
+			});
+
+		}
+
+		handleStateChange (newState) {
+			this.innerHTML = this.render(); // Refresh the HTML content
+		}
+
+		connectedCallback () {
+			console.log('GeneratorComponent#onConnectedCallback', this);
+			this.store.subscribe(this.onStateChange);
+		}
+
+		disconnectedCallback () {
+			console.log('GeneratorComponent#onDisconnectedCallback', this);
+			this.store.unsubscribe(this.onStageChange);
+		}
+
+		render () {
+			return `
+			<div class="generator-box">
+				<div class="generator-header">
+					<h3>${this.store.state.generators[this.dataset.id].name}</h3>
+					<h3>${this.store.state.generators[this.dataset.id].quantity}</h3>
+				</div>
+				<div class="generator-description">
+					<h4>${this.store.state.generators[this.dataset.id].description}</h4>
+				</div>
+				<div class="generator-footer">
+					<h4>${this.store.state.generators[this.dataset.id].rate}/60</h4>
+					<button class="generator-button">${this.store.state.generators[this.dataset.id].baseCost} STRAWBERRIES</button>
```

Comment:
instead of reading from `baseCost`, you should be reading cost from `getCost` function implemented from the generator in models.